geps/gep-2198.md

```diff
@@ -0,0 +1,105 @@
+
+JWT's can be encrypted, have methods for validation, can contain a variety of claims, and be used in gate keeping (validation) and authorization (claims matching) workflows.
+
+## How is a JWT used
+
+In practice a JWT is presented to an authorization layer to make a decision of whether or not to grant access to a specific hostname or application path (backend service) or also possibly used by the backend service to decide if specific secured data can be presented.
+This bring forth a set of common workflows which in turn present the use cases. In this example we will walk through the life of an encrypted JWT (though not all JWTs are encrypted)
+
+- client obtains a JWT
+- client presents the JWT with the request to a Gateway
+- the Gateway decrypts the JWT and validates it
+- now that the JWT is proven valid it is implicitly trusted and the request processing continues
+
+At this point a number of and/or paths can take place, most commonly a combination of the following happens:
+
+- the token content is tested for the existence of some claim property or value
+- claim values may be extracted and saved as header value(s) to the request
+- the entire token or a sub-set of claims may be forwarded directly to backend services for evaluation
+
+## Persona(s)
+
+The Personas that can be involved in managing JWT Policy can be:
+
+- platform / cluster operations
+- application developer / admins
+- user account manager
+- security operations
+
+## Prior related art
+
+Here are some samples of JWT implementations in a similar problem space.  Each implementation differs in terminology and implementation.
+
+- NGINX Ingress Controller:
+
+  - [https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-local-kubernetes-secret](https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-local-kubernetes-secret)
+  - [https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-jwks-from-remote-location](https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-jwks-from-remote-location)
+  
+- Kong:
+
+  - [https://docs.konghq.com/hub/kong-inc/jwt/configuration/](https://docs.konghq.com/hub/kong-inc/jwt/configuration/)
+
+- Envoy Gateway:
+
+  - [https://gateway.envoyproxy.io/latest/api/extension_types.html#jwtauthenticationfilterprovider](https://gateway.envoyproxy.io/latest/api/extension_types.html#jwtauthenticationfilterprovider)
+
+- Istio:
+
+  - [https://istio.io/latest/docs/tasks/security/authorization/authz-jwt/](https://istio.io/latest/docs/tasks/security/authorization/authz-jwt/)
+  - [https://istio.io/latest/docs/tasks/security/authentication/jwt-route/](https://istio.io/latest/docs/tasks/security/authentication/jwt-route/)
+  - [https://istio.io/latest/docs/tasks/security/authentication/claim-to-header/](https://istio.io/latest/docs/tasks/security/authentication/claim-to-header/)
+
+## Properties of a JWT Validation Policy
```

Comment:
I think it would be ideal to decouple JWT validation - as in 'is signature correct' ( same as mTLS handshake checks ) and the authorization parts ( which claims, paths, etc need to be matched to authorize the user ). 

For the later - it is best to have authorization decisions independent of the authn, and not mix it here.